react/features/filmstrip/components/web/Thumbnail.js

```diff
@@ -413,6 +416,25 @@ class Thumbnail extends Component<Props, State> {
      */
     componentDidMount() {
         this._onDisplayModeChanged();
+
+        // Listen to track streaming status changed event to keep it updated.
+        const { _videoTrack, dispatch } = this.props;
+        if (_videoTrack && !_videoTrack.local) {
+            _videoTrack.jitsiTrack.on(JitsiTrackEvents.TRACK_STREAMING_STATUS_CHANGED, this.handleTrackStreamingStatusChanged);
+            dispatch(trackStreamingStatusChanged(_videoTrack.jitsiTrack, _videoTrack.jitsiTrack.getTrackStreamingStatus?.()));
+        }
+    }
+
+    /**
+     * 
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentWillUnmount() {
+        const { _videoTrack, dispatch }  = this.props;
+        _videoTrack?.jitsiTrack.off(JitsiTrackEvents.TRACK_STREAMING_STATUS_CHANGED, this.handleTrackStreamingStatusChanged);
```

Comment:
can we also have this behind the source name flag?

Reply:
done